src/trusted/service_runtime/sel_mem.c

```diff
@@ -140,6 +140,30 @@ void NaClVmmapDtor(struct NaClVmmap *self) {
   self->vmentry = 0;
 }
 
+
+void NaClVmmapExitDtor(struct NaClVmmap *self) {
+  size_t i;
+
+  for (i = 0; i < self->nvalid; ++i) {
+    struct NaClVmmapEntry *entry;
+    entry = self->vmentry[i];
+
+    if (entry->desc != NULL) {
+        struct NaClDesc *ndp = entry->desc;
+        if (NULL != ndp) {
+          struct NaClRefCount *nrcp = (struct NaClRefCount *)&ndp->base;
+          struct NaClDescIoDesc *self = (struct NaClDescIoDesc *) nrcp;
+          free(self->hd);
+          self->hd = NULL;
+          free(nrcp);
+        }
+    }
```

Comment:
I have a feeling this could lead to double free issues, I think vmmap entries aren't guaranteed to be the only things holding certain nacl descs, particularly for mmap'ed files.